modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -3095,8 +3095,6 @@ def _arrow_type_to_dtype(cls, arrow_type):
         except NotImplementedError:
             if pyarrow.types.is_time(arrow_type):
                 res = np.dtype(datetime.time)
-            elif pyarrow.types.is_dictionary(arrow_type):
```

Comment:
Do we have a similar problem for PandasOn<Ray/Dask>? If yes, please create an issue for this.